### PR TITLE
Modal Component

### PR DESCRIPTION
**Pull request by daniel-sm-yu**
## Notion ticket link

[Builder - Modal Component](https://www.notion.so/uwblueprintexecs/Task-Board-485f846ae00243599dddba2be9bcba82?p=22e6c1a2c2b345c9971cdd884f7467f0)

## Implementation description
* Create a Modal component using Chakra UI
* Configure Modals to support child components
* Make Modal customizable 

## Steps to test
```
    const { isOpen, onOpen, onClose } = useDisclosure();
    ...
    <Button onClick={onOpen}>Open Modal</Button>
    <Modal
        isOpen={isOpen}
        onCancel={onClose}
        onConfirm={onClose}
        header="Delete [MODULE]"
        bodyText="Are you sure?"
        confirmText="Delete"
        confirmButtonColor="red"
        size="lg"
    />
```
image
```
    const { isOpen, onOpen, onClose } = useDisclosure();
    ...
    <Button onClick={onOpen}>Open Modal</Button>
    <Modal
        isOpen={isOpen}
        onCancel={onClose}
        onConfirm={onClose}
        header="Link [MODULE]"
        alignButtonsRight={false}
    >
        <Button bg="grey">[this can be anything]</Button>
    </Modal>
```
image

## What should reviewers focus on?
* 

## Checklist
- [ ] My PR name is descriptive and in imperative tense
- [ ] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [ ] I have run the appropriate linter(s)
- [ ] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

**Review by mkanoria — APPROVED** [review]
This looks great!! Thank you for building this and also writing the example on how to use the component - that's very helpful! 

Let's keep in mind to merge this after @tawnyzhao's PR is merged in!

File: src/components/modal/index.tsx
Comment (tawnyzhao) [inline src/components/modal/index.tsx:18]: Consider making `header` to `header?` and give it a default value of `""` to make it optional in case the modal just needs to be a custom component.

File: src/components/modal/index.tsx
Comment (tawnyzhao) [inline src/components/modal/index.tsx:41]: Rather than having only a few options to override like size, consider adding a `...rest` to forward props to `ChakraModal` in case we need to change the props here https://chakra-ui.com/docs/overlay/modal#props.

**Review by tawnyzhao — APPROVED** [review]

File: src/components/modal/index.tsx
Comment (daniel-sm-yu) [reply src/components/modal/index.tsx]: great point!

**Pull request merged**
